data_collection/gazette/spiders/sp_sorocaba.py

```diff
@@ -0,0 +1,100 @@
+import re
+from datetime import date, datetime
+
+from scrapy import Request
+from dateparser import parse
+from dateutil.rrule import MONTHLY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpSorocabaSpider(BaseGazetteSpider):
+    name = "sp_sorocaba"
+    TERRITORY_ID = "3552205"
+
+    start_date = date(2005, 1, 7)
+
+    gazettes_list_url = "http://noticias.sorocaba.sp.gov.br/jornal/?ano={}"
+
+    MonthDict = {
+        1: "Janeiro",
+        2: "Fevereiro",
+        3: "Marco",
+        4: "Abril",
+        5: "Maio",
+        6: "Junho",
+        7: "Julho",
+        8: "Agosto",
+        9: "Setembro",
+        10: "Outubro",
+        11: "Novembro",
+        12: "Dezembro",
+    }
+
+    def start_requests(self):
+        end_date = date.today()
+
+        periods_of_interest = [
+            (date.year, date.month)
+            for date in rrule(freq=MONTHLY, dtstart=self.start_date, until=end_date)
+        ]
+        global year

```

Comment:
**question**: Why do you need this global variable?

**issue**: Scrapy is asynchronous, so you will never be sure about the value of this variable as it may change in a way we don't expect. There are better and safer ways to pass the year between requests (using [Request.meta](https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta) or [Request.cb_kwargs](https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.cb_kwargs) for example).